Service/Client/ClientFactory.php

```diff
@@ -56,15 +57,15 @@ public function create(int $storeId = 0, ?array $data = []): ?ClientInterface
      * @return ClientInterface|null
      * @throws SignerException
      */
-    private function createClient(array $credentials): ?ClientInterface
+    private function createClient(array $credentials, ?bool $forceSandbox = null): ?ClientInterface
     {
         Settings::tlAgent('truelayer-magento/' . $this->configProvider->getExtensionVersion());
         return Client::configure()
             ->clientId($credentials['client_id'])
             ->clientSecret($credentials['client_secret'])
             ->keyId($credentials['key_id'])
             ->pemFile($credentials['private_key'])
-            ->useProduction(!$this->configProvider->isSandbox())
+            ->useProduction(is_null($forceSandbox) ? !$this->configProvider->isSandbox() : !$forceSandbox)
```

Comment:
Can this be simplified to `!$forceSandbox && !$this->configProvider->isSandbox())`?

Reply:
That would have the following result: when $forceSandbox=false it would fall back to saved config. $forceSandbox defaults to null, the "check credentials" is the only place where it's set to explicitly use the environment from the form value. Here's when this slight change would actually cause a problem:
Say you are done testing in sandbox. You want to switch to production, fill out the form, and you click "Check Credentials" without saving the form. $forceSandbox is false, so it will fall back to the saved environment, which is still sandbox.